backend/inspirehep/pidstore/minters/cnum.py

```diff
@@ -0,0 +1,109 @@
+
+
+class CNUMMinter(Minter):
+
+    pid_value_path = "cnum"
+    pid_type = "cnum"
+    provider = InspireCNUMProvider
+
+    @classmethod
+    def mint(cls, object_uuid, data):
+        """Mint a CNUM identifier for a conference record
+
+        This method calculates the next CNUM identifier according to the
+        conference schema. In case a CNUM is already present in the metadata,
+        the minter registers a new pid for it. This can happen when conference
+        records are migrated from legacy.
+
+        This method doesn't handle the case where conference records are created
+        with the flag `deleted` set to True.
+
+        Args:
+            object_uuid (str): the record uuid
+            data (dict): the record's metadata
+
+        Returns:
+            Minter: an instance of this class used for minting the CNUM.
+        """
+        minter = cls(object_uuid, data)
+        minter.validate()
+        if "cnum" not in data:
+            cnum_provider = minter.create(data)
+            if cnum_provider:
+                cnum = cnum_provider.pid.pid_value
+                data["cnum"] = cnum
+        else:
+            # migrated record already have a CNUM identifier in metadata
+            PersistentIdentifier.create(
+                pid_type="cnum",
+                pid_value=data["cnum"],
+                object_uuid=object_uuid,
+                object_type="rec",
+                status=PIDStatus.REGISTERED,
+            )
+        return minter
+
+    @classmethod
+    def update(cls, object_uuid, data):
+        """Handle the CNUM indentifier in record updates .
+
+        This method performs several checks:
+        * if it is the first update for this record, don't do anything;
+        * if the CNUM has been modified raises an error;
+        * if this record gets un-deleted then un-delete also its CNUM pid.
+
+        Args:
+            object_uuid (str): the record uuid
+            data (dict): the record's metadata
+
+        Returns:
+            None
+        """
+        from inspirehep.records.api import ConferencesRecord
+
+        rec = ConferencesRecord.get_record_by_pid_value(data['control_number'])
+        try:
+            prev_version = rec._previous_version
+        except AttributeError:
+            return
+
+        changed_cnum = rec.get("cnum") != prev_version.get("cnum")
+        if changed_cnum:
+            raise CNUMChanged(f"CNUM identifier changed from the prev version")
+
+        changed_deleted_status = rec.get("deleted", False) ^ prev_version.get(
+            "deleted", False
+        )
+        if changed_deleted_status:  # record is un-deleted
+            pid = PersistentIdentifier.query\
+                .filter(PersistentIdentifier.pid_type == "cnum")\
+                .filter(PersistentIdentifier.object_uuid == rec.id)\
+                .one_or_none()
+            if pid:
+                pid.status = PIDStatus.REGISTERED  # un-delete pid
```

Comment:
this whole thing with versions seems overly complex. I would do the following:
1. Get the cnum pointing to this record from the pidstore
2. Is it the `pid_value` compatible with the metadata? if not -> exception
3. If the record is not marked as `deleted` but the `pid` is deleted, mark it as `registered`.

Reply:
Totally agree